quarkus.jdt/com.redhat.quarkus.jdt.core/src/main/java/com/redhat/quarkus/jdt/core/JDTQuarkusManagerForJava.java

```diff
@@ -0,0 +1,351 @@
+			}
+			try {
+				return this.applicationPropertiesFile = javaProject.getProject().getLocation().removeLastSegments(1)
+						.append(javaProject.getOutputLocation()).append(APPLICATION_PROPERTIES_FILE).toFile();
+			} catch (JavaModelException e) {
+				return null;
+			}
+		}
+
+		/**
+		 * Returns the loaded application.properties and null otherwise.
+		 * 
+		 * @return the loaded application.properties and null otherwise
+		 * @throws JavaModelException
+		 */
+		public Properties getApplicationProperties() throws JavaModelException {
+			File applicationPropertiesFile = getApplicationPropertiesFile();
+			if (applicationPropertiesFile == null) {
+				reset();
+				return null;
+			}
+			if (applicationPropertiesFile.lastModified() != lastModified) {
+				reset();
+				try (InputStream input = new FileInputStream(applicationPropertiesFile)) {
+					properties = new Properties();
+					// load a properties file
+					properties.load(input);
+				} catch (IOException e) {
+					reset();
+					LOGGER.log(Level.SEVERE,
+							"Error while loading properties from '" + applicationPropertiesFile.getPath() + "'.", e);
+				}
+			}
+			return properties;
+		}
+
+		private void reset() {
+			properties = null;
+			serverPort = null;
+		}
+
+		/**
+		 * Returns the server port.
+		 * 
+		 * @return the server port.
+		 */
+		public int getServerPort() {
+			try {
+				try {
+					if (serverPort == null) {
+						Properties properties = getApplicationProperties();
+						if (properties != null) {
+							String port = properties.getProperty("quarkus.http.port", "");
+							if (!port.trim().isEmpty()) {
+								serverPort = Integer.parseInt(port.trim());
+							}
+						}
+					}
+				} catch (JavaModelException e) {
+					LOGGER.log(Level.SEVERE, "Error while getting 'quarkus.http.port", e);
+				}
+			} finally {
+				if (serverPort == null) {
+					serverPort = DEFAULT_PORT;
+				}
+			}
+			return serverPort;
+		}
+	}
+
+	public static JDTQuarkusManagerForJava getInstance() {
+		return INSTANCE;
+	}
+
+	private final Map<IJavaProject, JDTQuarkusProjectInfo> infos;
+
+	private JDTQuarkusManagerForJava() {
+		this.infos = new HashMap<>();
+	}
+
+	/**
+	 * Returns the codelens list according the given codelens parameters.
+	 * 
+	 * @param params  the codelens parameters
+	 * @param utils   the utilities class
+	 * @param monitor the monitor
+	 * @return the codelens list according the given codelens parameters.
+	 * @throws JavaModelException
+	 */
+	public List<? extends CodeLens> codeLens(QuarkusJavaCodeLensParams params, IJDTUtils utils,
+			IProgressMonitor monitor) throws JavaModelException {
+		String uri = params.getUri();
+		ITypeRoot typeRoot = resolveTypeRoot(uri, utils, monitor);
+		if (typeRoot == null) {
+			return Collections.emptyList();
+		}
+		IJavaElement[] elements = typeRoot.getChildren();
+		Collection<CodeLens> lenses = new LinkedHashSet<>(elements.length);
+		if (params.isUrlCodeLensEnabled()) {

```

Comment:
check for server available should happen here, only once

Reply:
If I do that I check the server for any Java file. In my case I check the server only once for Java class bound with JAX-RS Path (just for the class not for the method), that's to say, the server check is done once.

Reply:
there's no need to iterate on all members if the server is not even started

Reply:
> there's no need to iterate on all members if the server is not even started

I would like to avoid checking the server state of Java class in not a JAX-RS (annotated with @Path).

Given this java file:
```
@Path
public class A {

  public void f()

  public void g()
}
```

One check is done (only on A java element IType not for the methods)


Given this java file

```
class B {

}
```

None check on server state is done since B is no bound to @Path.